File: csharp/cars-assemble/CarsAssemble.cs
using System;

static class AssemblyLine
{
    public static double SuccessRate(int speed)
    {
        if (speed >= 1 && speed <= 4)
        {
            return 1;
        }
        if (speed >= 5 && speed <= 8)
        {
            return .9;
        }
        if (speed == 9)
        {
            return .8;
        }
        if (speed == 10)
        {
            return .77;
        }
        return 0;
    }

    public static double ProductionRatePerHour(int speed)
    {
        return SuccessRate(speed) * 221 * speed;
    }

    public static int WorkingItemsPerMinute(int speed)
    {
        return (int)(ProductionRatePerHour(speed) / 60);
    }
}

File: java/cars-assemble/src/main/java/CarsAssemble.java
public class CarsAssemble {

    public double productionRatePerHour(int speed) {
        int fullEfficiencyProductionRate = 221 * speed;
        switch (speed) {
            case 10: {
                return fullEfficiencyProductionRate * 0.77;
            }
            case 9: {
                return fullEfficiencyProductionRate * 0.8;
            }
            case 8:
            case 7:
            case 6:
            case 5: {
                return fullEfficiencyProductionRate * 0.9;
            }
            case 4:
            case 3:
            case 2:
            case 1: {
                return fullEfficiencyProductionRate;
            }
            default: {
                return 0;
            }
        }
    }

    public int workingItemsPerMinute(int speed) {
        return (int) (productionRatePerHour(speed) / 60);
    }
}
